### PR TITLE
Exclude Python 3.9.7 due to import error in catalog module

### PR DESCRIPTION
**Pull request by ndrluis**
Adjusts the supported Python versions to exclude 3.9.7, as importing the catalog module on this version results in an unexpected error. The issue does not occur in versions 3.9.6 or 3.9.8, indicating a specific incompatibility with Python 3.9.7.

Related to #500, #497

### CODE REVIEW

**Comment by ndrluis** [comment]
Related python issue: https://bugs.python.org/issue45081

**Comment by sungwy** [comment]
This sounds like a great idea @ndrluis - thank you for identifying this issue and putting in the fix. Could we ask for you to rebase your PR against the current main, and run `poetry lock` with your proposed python version change?

**Comment by ndrluis** [comment]
@sungwy Done!

**Comment by kevinjqliu** [comment]
With this change, what do I see as a user with python 3.9.7? Does it show that 3.9.7 is not supported and I should use a supported version?

**Comment by ndrluis** [comment]
Yes @kevinjqliu! 

This is a example
image

**Review by sungwy — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -49,7 +49,7 @@ include = [
 ]
 
 [tool.poetry.dependencies]
-python = "^3.8"
+python = ">=3.8,<3.9.7 || >=3.9.8,<4.0"
```
Comment (kevinjqliu) [inline pyproject.toml:52]: nit: 
```suggestion
python = ">=3.8, !=3.9.7"
```

See [Multiple Version Specifiers](https://stackoverflow.com/questions/54720072/dependency-version-syntax-for-python-poetry)

Comment (ndrluis) [reply pyproject.toml]: I made the change, but I needed to add <4.0 to !=3.9.7 in order for it to work.
image
Now the error message is easier to understand! Thank you @kevinjqliu!

**Review by kevinjqliu — APPROVED** [review]
LGTM! 
I've tested with python 3.9.7 and running `make install`
Here's the error log
```
The currently activated Python version 3.9.7 is not supported by the project (>=3.8, <4.0, !=3.9.7).
Trying to find and use a compatible version.
```

**Pull request merged**